plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -235,21 +236,21 @@
     @Override
     public void finalizeSnapshot(SnapshotId snapshotId, ShardGenerations shardGenerations, long startTime, String failure, int totalShards,
                                  List<SnapshotShardFailure> shardFailures, long repositoryStateId, boolean includeGlobalState,
-                                 MetaData clusterMetaData, Map<String, Object> userMetadata, boolean writeShardGens,
+                                 MetaData clusterMetaData, Map<String, Object> userMetadata, Version version,
```

Comment:
can we give this a more meaningful name than just "version"? Perhaps `repositoryMetaVersion`?